llvm/lib/MCCAS/MCCASObjectV1.cpp

```diff
@@ -3090,9 +3090,26 @@ struct DIEVisitor {
   Error visitDIERef(DIEDataRef Ref);
   Error visitDIERef(BinaryStreamReader &DataReader, unsigned AbbrevIdx,
                     StringRef DIEData, ArrayRef<DIEDataRef> &DIEChildrenStack);
-  Error visitDIEAttrs(AbbrevEntryReader &AbbrevReader,
-                      BinaryStreamReader &Reader, StringRef DIEData);
+  Error visitDIEAttrs(BinaryStreamReader &Reader, StringRef DIEData,
+                      unsigned AbbrevIdx);
+  Error materializeAbbrevDIE(unsigned AbbrevIdx);
+
+  struct AbbrevDIEContents {
+    dwarf::Attribute Attr;
+    dwarf::Form Form;
+    bool FormInDistinctData;
+    bool FormSizeRequiresDIEContents;
+    uint8_t FormSize;
```

Comment:
`FormSizeRequiresDIEContents` and `FormSize` are mutually exclusive. Let's combine the two into an `optional<uint8_t> FormSize`.

Reply:
Ahh yes, good idea!